### ISSUE TITLE
Verification does not work as expected

### ISSUE THREAD

**Issue by anotherbridge** [in-toto/in-toto-golang#207]
**General information:**

I did compare the Python and Golang versions of the in-toto-framework. In order to do so I followed the [demo-project](https://github.com/in-toto/demo). 
During the verifification step the golang version seemed to have generated a corrupt `untar.link` file. 

* Version (tag or commit): checked out repo at state of commit 82bb771212f4cfb71da8a325278f3cce6f042205
* Operating system (Linux/Mac/Windows): macOS Ventura (ARM architecture)
* Go version: `go version go1.19.6`
* Go build flags (if any):

**Description of the bug:**

When building the in-toto binary I ran: 

```
make build 
make test
```

According to the output all the tests seemed to be successful. Then I followed all the steps described in the demo project. All steps worked as expected when using the in-toto Golang version (denoted as `in-toto-golang`) until the point of verification. When running the verification step something interesting happened: 

1. Running it it with in the Python version `in-toto-verify --layout root.layout --layout-key alice.pub` resulted in a success and return status 0. 
2. Then running the go version `in-toto-golang verify --layout root.layout --layout-keys alice.pub` resulted in the following output: 
```
inspection failed: artifact verification failed for Inspection 'untar', materials [clone.776a00e2.link package.2f89b927.link untar.link demo-project/foo.py update-version.776a00e2.link] disallowed by rule [DISALLOW *]
```
3. After that I tried running the verification of the python version (with the same `untar.link` as generated by the go version) again and got the following result back: 
```
(in-toto-verify) RuleVerificationError: 'DISALLOW *' matched the following artifacts: ['demo-project/foo.py']
Full trace for 'expected_materials' of item 'untar':
Available materials (used for queue):
['.keep', 'alice.pub', 'demo-project.tar.gz', 'root.layout', 'demo-project/foo.py']
Available products:
['.keep', 'alice.pub', 'demo-project.tar.gz', 'root.layout', 'demo-project/foo.py']
Queue after 'MATCH demo-project.tar.gz WITH PRODUCTS FROM package':
['root.layout', '.keep', 'alice.pub', 'demo-project/foo.py']
Queue after 'ALLOW .keep':
['root.layout', 'alice.pub', 'demo-project/foo.py']
Queue after 'ALLOW alice.pub':
['root.layout', 'demo-project/foo.py']
Queue after 'ALLOW root.layout':
['demo-project/foo.py']
```

The two versions of the `untar.link` file are given as follows: 
[Golang](https://github.com/in-toto/in-toto-golang/files/10765254/untar.link.golang.txt)
[Python](https://github.com/in-toto/in-toto-golang/files/10765256/untar.link.python.txt)

P.S. I had to add a `.txt` extension to both of the files, since GitHub wouldn't let me upload them otherwise.

**Comment by adityasaky** [comment]
Hi @anotherbridge, this is caused because the first time you ran verification, it created some artifacts (it untar-ed the archive and created link metadata for the inspection). Subsequent verification attempts record these artifacts and consider them invalid. Can you try your comparison in two clean copies of the demo repo? Thanks!

**Comment by anotherbridge** [comment]
Hey, thanks for the response @adityasaky. 

Unfortunately when running the same in two different and clean copies of the demo repo the same occurs. For the Python version everything works fine, but when running the verification with the Go version I'm still getting the same error: 

```
in-toto verify --layout root.layout --layout-keys alice.pub

inspection failed: artifact verification failed for Inspection 'untar', materials [update-version.776a00e2.link clone.776a00e2.link package.2f89b927.link] disallowed by rule [DISALLOW *]
```

This was the directory structure before I ran the verification in **both** cases: 
```
.
├── alice.pub
├── clone.776a00e2.link
├── demo-project.tar.gz
├── package.2f89b927.link
├── root.layout
└── update-version.776a00e2.link
```

Again here are the `untar.link` files generated in both cases: 
[Golang](https://github.com/in-toto/in-toto-golang/files/10791491/untar.link.golang.txt)
[Python](https://github.com/in-toto/in-toto-golang/files/10791495/untar.link.python.txt)

**Comment by adityasaky** [comment]
Good catch, @anotherbridge! Here's why this is happening: https://github.com/in-toto/in-toto/blob/fbc1eb053dd39c6d1fb307a60bd2c781e667c931/in_toto/settings.py#L39

The Python implementation automatically excludes recording certain artifacts which the Go implementation does not. @lukpueh and @shibumi, do we add these defaults to the Go implementation?

**Comment by anotherbridge** [comment]
Thanks very much @adityasaky! That explains the behavior. And when explicitly allowing the `*.link` files, the verification does work as expected.

**Comment by adityasaky** [comment]
Great! I'm going to edit this ticket to record exactly why this is happening.

**Comment by adityasaky** [comment]
I'm actually less sure if we should patch it here or in in-toto-python. IMO the default excludes is not immediately obvious and potentially unexpected behaviour to a newcomer. @lukpueh WDYT?
